platform/engineering/design-docs/separate-content-build.md

```diff
@@ -0,0 +1,370 @@
+  command, which isn't obvious
+- **Application deployments and PR builds fail** when content in the CMS is
+  invalid
+  - This includes broken links and accessibility violations
+  - We have a safeguard that mostly works; if we run into a failure using the
+    latest CMS content, we try again with the cached content in S3 from the last
+    successful build
+    - This doesn't work when the GraphQL query has been updated; that
+      invalidates the cache
+    - Errors in liquid templates would potentially break the build and falling
+      back to the cache won't help
+
+### High Level Design
+The current front end build will be split up into two distinct builds:
+1. Content build
+    - Input: CMS content
+    - Output: HTML files, assets from the CMS (PDFs, images and the like)
+    - This maps pretty cleanly to the current partial deploy
+1. Application build
+    - Input: `vets-website` code
+    - Output: JavaScript and CSS bundles
+    - This essentially maps to the current full deploy minus the content
+
+The output of both these builds will be deployed to a single S3 bucket with
+safeguards in place to ensure they don't override each other.
+
+**Another important note:** There is no way to coordinate these two deploys to
+make an application live for the first time. The process will be to manually:
+1. Verify the application assets are live in production
+1. Make the landing page live in production in the content build
+    - Depending on the answer to [this
+      question](#react-application-landing-pages), this may mean publishing the
+      page in Drupal or switching the flag in the JSON file
+1. Deploy a new content build
+
+## Specifics
+
+### Detailed Design
+
+#### Content build in a separate repository
+The content build will live in a separate repository called `content-build`.
+This repository will contain only code pertaining to the content build.
+
+![Separated content code base](images/phase-2-separated-content-build.png)
+
+#### Coordinating deployments to S3
+We use S3 to serve a static site. To do this, the bucket consists of a file tree
+which corresponds to the URLs on VA.gov. For example, `/education/index.html` is
+the file that's served at www.va.gov/education/.
+
+The output of the application build **currently** live in the `/generated/`
+directory. To support the transition, the **new** application build will put its
+assets in the `/applications/` directory.
+
+![Coordinating deployments to S3](images/coordinating-deployments-to-s3.png)
+
+To ensure that one build won't override files in the other, we'll have to
+1. Update
+   [`vets-website-deploy.sh`](https://github.com/department-of-veterans-affairs/devops/blob/2c63e6d62fa5ad2b6b87478e30d0a7bc2820c252/ansible/deployment/roles/deploy-vets-website/files/vets-website-deploy.sh)
+   to copy only to the `/applications/` directory
+1. Create a new deploy script for the content deployment which copies everything
+   over _except_ the `/applications/` directory
```

Comment:
This seems not quite right, but....I don't remember the specifics. :bat-signal: @nathanhruby

Reply:
Also, would this be a good place to add the bit about updating the "poor man's CDN" (as you so eloquently described it the other day :smile: )? Looking at the script I linked to, it kind of reads like that's the one that copies it to the assets bucket, but....I'm not 100% sure and would appreciate any clarification you can lend.